test/functional/wallet_hd.py

```diff
@@ -118,5 +119,38 @@ def run_test (self):
 
         assert_equal(keypath[0:7], "m/0'/1'")
 
+        # Generate a new HD seed on node 1 and make sure it is set
+        orig_masterkeyid = self.nodes[1].getwalletinfo()['hdmasterkeyid']
+        self.nodes[1].sethdseed()
+        new_masterkeyid = self.nodes[1].getwalletinfo()['hdmasterkeyid']
+        assert orig_masterkeyid != new_masterkeyid
+        addr = self.nodes[1].getnewaddress()
+        assert_equal(self.nodes[1].getaddressinfo(addr)['hdkeypath'], 'm/0\'/0\'/0\'') # Make sure the new address is the first from the keypool
```

Comment:
nit: also check it's not the same key(not just path)

Reply:
Done